src/utils/withWidth.js

```diff
@@ -35,15 +37,19 @@ const withWidth = (options = {}) => Component => {
     };
 
     componentDidMount() {
-      this.updateWidth(window.innerWidth);
+      const win = ownerWindow(findDOMNode(this.mountNode));
+      this.updateWidth(win.innerWidth);
     }
 
     componentWillUnmount() {
       this.handleResize.cancel();
     }
 
+    mountNode = null;
+
     handleResize = debounce(() => {
-      this.updateWidth(window.innerWidth);
+      const win = ownerWindow(findDOMNode(this.mountNode));
```

Comment:
This appears to work better. I'm still not 100% confident in its function.

`findDOMNode(this.mountNode)` always returns null from `componentDidMount` when a `width` or `initialWidth` prop isn't provided (such as in the docs), since we render `null` when width isn't defined yet in props or state.